### PR TITLE
docs: add proofs queries documentation

### PR DESCRIPTION
**Pull request by SweeXordious**
## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]

File: docs/developers/blobstream-proof-query.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+This [endpoint](https://github.com/celestiaorg/celestia-core/blob/793ece9bbd732aec3e09018e37dc31f4bfe122d9/rpc/core/tx.go#L175-L213) allows querying a shares proof to row roots, then a row roots to data root proofs. It takes a block `height`, a starting share index and an end share index which define a share range. Then, two proofs are generated:
+
+- An NMT proof of the shares to the row roots
+- A binary merkle proof of the row root to the data root
+
+> **_NOTE:_** if the share range spans multiple rows, then the proof can contain multiple NMT and binary proofs.
+
+Example request: `/prove_shares?height=15&startShare=0&endShare=1`
+
+Which queries the proof of shares `[0,1)` in block `15`.
```
Comment (musalbas) [inline docs/developers/blobstream-proof-query.md:67]: should this be a ]?

**Comment by musalbas** [comment]
I assume we need to add the new page to sidebar

File: docs/developers/blobstream-proof-query.md (hunk 1 above)
Comment (SweeXordious) [reply docs/developers/blobstream-proof-query.md:67]: this is meant to be end exclusive. I used `[` in documentation before but apparently `)` is more used to signify end exclusivity

**Comment by SweeXordious** [comment]
> I assume we need to add the new page to sidebar

Done

**Review by jcstein — APPROVED** [review]

**Pull request merged**